script/tool/lib/src/version_check_command.dart

```diff
@@ -96,48 +94,59 @@ class VersionCheckCommand extends PluginCommand {
     final List<String> changedPubspecs =
         await gitVersionFinder.getChangedPubSpecs();
 
-    final String baseSha = argResults[_kBaseSha] as String;
```

Comment:
I still think we should allow explicit base-sha so this can be used in post-submit where we can set base-sha to be HEAD~

Reply:
We do; all the support for that flag is in the common.dart code. See the addition of _getBaseSha when null is passed, which uses the existing (working) logic to either use the flag, or the merge base when not passed.

Most of the tests use the flag, and still pass.

Reply:
(Also, we should definitely fix the post-submit invocation since currently I believe it does nothing. I'll do that in a follow-up. And longer term we should maybe reconsider how this check works, since it ignores the usual packages/all system.)

Reply:
ah i actually developed that code not even long time ago lol my memory is getting really bad these days due to lack of sleep. :)